### PR TITLE
Add label semantic examples with new Fluid api

### PR DESCRIPTION
**Pull request by sidgoyal78**
Issues:
- This PR doesn't take into account the evaluator (https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/test_label_semantic_roles.py#L185)
- There is some code duplication, in functions `train_network()` and `inference_network()`.

### CODE REVIEW

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    feature_out = fluid.layers.sums(input=[
+        fluid.layers.fc(input=input_tmp[0], size=label_dict_len, act='tanh'),
+        fluid.layers.fc(input=input_tmp[1], size=label_dict_len, act='tanh')
+    ])
+
+    return feature_out
+
+
+def inference_network():
+    word = fluid.layers.data(
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Can we put data layer definitions into `lstm_net`?

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+import numpy
+
+word_dict, verb_dict, label_dict = paddle.dataset.conll05.get_dict()
+word_dict_len = len(word_dict)
+label_dict_len = len(label_dict)
+pred_dict_len = len(verb_dict)
+mark_dict_len = 2
+
+def lstm_net(word, predicate, ctx_n2, ctx_n1, ctx_0, ctx_p1, ctx_p2, mark,
+            **ignored):
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Why do we need `**ignored`?

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    predict = lstm_net(word, predicate, ctx_n2, ctx_n1, ctx_0, ctx_p1, ctx_p2,
+                  mark)
+ 
+    crf_decode = fluid.layers.crf_decoding(
+        input=feature_out, param_attr=fluid.ParamAttr(name='crfw'))
+
+    return crf_decode
+
+def train_network():
+    mix_hidden_lr = 1e-3
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Some of the constants in this file is upper case snake case (BATCH_SIZE), some are snake case (mix_hidden_lr). Could you change all constants to upper case snake case (BATCH_SIZE)?

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    def event_handler(event):
+        if isinstance(event, fluid.EndIteration):
+            if (event.batch_id % 10) == 0:
+                avg_cost = trainer.test(reader=test_reader)
+
+                print('BatchID {1:04}, Loss {2:2.2}'.format(
+                    event.batch_id + 1, avg_cost))
+
+                if avg_cost > 0.01:  # Low threshold for speeding up CI
+                    trainer.params.save(save_path)
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Sorry that the API changed a little, please use `trainer.save_params` (https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/word2vec/no_test_word2vec_new_api.py#L103)

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+        optimizer=sgd_optimizer,
+        place=place,
+        event_handler=event_handler)
+    trainer.train(train_reader, EPOCH_NUM, event_handler=event_handler)
+
+
+def infer(use_cuda, save_path):
+    params = fluid.Params(save_path)
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+    inferencer = fluid.Inferencer(inference_network, params, place=place)
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Sorry that the API changed a little (param.py is removed), please use 
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/word2vec/no_test_word2vec_new_api.py#L108 and https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/word2vec/no_test_word2vec_new_api.py#L118

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py (hunk 1 above)
Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Done.

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py (hunk 2 above)
Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Removed, thanks.

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py (hunk 3 above)
Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Done, thanks.

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py (hunk 4 above)
Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Done

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py (hunk 5 above)
Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Done.

File: python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+        batch_size=BATCH_SIZE)
+    test_reader = paddle.batch(
+        paddle.dataset.conll05.test(), batch_size=BATCH_SIZE)
+
+    def event_handler(event):
+        if isinstance(event, fluid.EndIteration):
+            if (event.batch_id % 10) == 0:
+                avg_cost = trainer.test(reader=test_reader)
+
+                print('BatchID {1:04}, Loss {2:2.2}'.format(event.batch_id + 1,
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: I think it should be `BatchID {0:04}, Loss {1:2.2}`

Comment (sidgoyal78) [reply python/paddle/fluid/tests/book/label_semantic_roles/no_test_label_semantic_roles.py]: Done

**Pull request merged**